presto-spi/src/main/java/com/facebook/presto/spi/type/StructuralType.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.LazyBlockLoader;
+
+/**
+ * StructuralType is a type that are structural (array, map, row) and contain other blocks inside it
+ */
+public interface StructuralType
+        extends Type
+{
+    public Block createLazyBlock(int positionCount, LazyBlockLoader loader);
```

Comment:
Why do we need this? It looks like every implementation is the same. I also don't think this should be part of the `Type` API